### PR TITLE
Add a new object type DataLoopNode::ConnectionObjectType

### LINKED ISSUE #10204: Add more object types in the DataLoopNode::ConnectionObjectType
Issue overview
--------------
The NodeInputManager::GetOnlySingleNode function is used to get a node number. One of requirements is to provide DataLoopNode::ConnectionObjectType. The current list of ConnectionObjectType may not cover all possible object types, such as AirLoopHVAC:Splitter and AirLoopHVAC:Mixer. These two objects may be needed in the list, so that the GetOnlySingleNode can be used to get node numbers defined in these two objects.

This issue was raised after review of PR #10042

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
 - Fixes #10204 
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

A correct function is required to get an air node number. The function requires a connection type as an argument. The AirLoopHVAC:Splitter object is not available in the connection list. The fix adds the object in the list.

The test file is an example file as SmallOffice_CentralDOAS.idf.    

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ X] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ X] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 1]
```diff
@@ -383,7 +383,15 @@ namespace AirLoopHVACDOAS {
 
                 thisSplitter.name = Util::makeUPPER(thisObjectName);
                 thisSplitter.InletNodeName = Util::makeUPPER(fields.at("inlet_node_name").get<std::string>());
-                thisSplitter.InletNodeNum = Util::FindItemInList(thisSplitter.InletNodeName, state.dataLoopNodes->NodeID);
+                thisSplitter.InletNodeNum = NodeInputManager::GetOnlySingleNode(state,
```
Comment (lgu1234) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:394]: The correct function is used to get an air node number.

File: tst/EnergyPlus/unit/AirLoopHVACDOAS.unit.cc [hunk 2]
```diff
@@ -11579,6 +11579,9 @@ TEST_F(EnergyPlusFixture, AirLoopHVACDOAS_TestFanDrawThroughPlacement)
     EXPECT_NEAR(AirLoopDOAS_SplitterInletNode.MassFlowRate, 0.1, 0.0001);
     EXPECT_NEAR(DOAS_FanInletNode.MassFlowRate, 0.1, 0.0001);
     EXPECT_NEAR(DOAS_HeatingCOilOutletNode.MassFlowRate, 0.1, 0.0001);
+
+    // Issue 10204 Get AirLoopHVAC:Splitter inlet node number using NodeInputManager::GetOnlySingleNode
+    EXPECT_EQ(thisAirLoopDOASObjec.m_CompPointerAirLoopSplitter->InletNodeNum, 17);
```
Comment (lgu1234) [inline tst/EnergyPlus/unit/AirLoopHVACDOAS.unit.cc:11584]: Unit test

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 1 above)
Comment (mjwitte) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:394]: @lgu1234 This looks good. The same change is needed for the Mixer inlet nodes and the Splitter outlet nodes.

Comment (lgu1234) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:394]: @mjwitte Thanks. The same changes are made.

**Review by mjwitte — APPROVED** [review]
@lgu1234 Looks good. Unit tests run, diffs are as expected. Merging.

**Pull request merged**